### PR TITLE
[IMP][13.0] web_responsive

### PR DESCRIPTION
**Pull request by ivantodorovich**
Continuing after https://github.com/OCA/web/pull/1654, a few more style improvements

image

image

----

Also removed `display: flex` on search results because it was causing this issue:

**Pre:**
pre-display-flex

**Post:**
post-display-flex

### CODE REVIEW

File: web_responsive/static/src/css/web_responsive.scss [hunk 1]
```diff
@@ -181,23 +177,31 @@ $chatter_zone_width: 35%;
...
         }
 
         .o_app {
             align-items: center;
             display: flex;
+            padding: 10px 0;
             flex-direction: column;
             justify-content: flex-start;
             background: none;
+            transition: all 0.3s ease 0s;
```
Comment (yajo) [inline web_responsive/static/src/css/web_responsive.scss]: Remember that `transition: all` is dangerous. Could you be more specific?

**Review by yajo — CHANGES_REQUESTED** [review]
One change suggested.

Also, I see a big usage of hardcoded colors and px sizes. Please  use color variables from the theme and rem untis wherever possible.

File: web_responsive/static/src/css/web_responsive.scss (hunk 1 above)
Comment (ivantodorovich) [reply web_responsive/static/src/css/web_responsive.scss]: Done, thanks!

**Comment by ivantodorovich** [comment]
@Yajo what about now?

Not sure what you meant about hardcoded colors... they're just black/white with opacity. But I changed the rgb values for the keyword.. is that what you meant?

File: web_responsive/static/src/css/web_responsive.scss [hunk 2]
```diff
@@ -181,23 +179,32 @@ $chatter_zone_width: 35%;
...
+            transition-property: background-color;
+
             img {
                 box-shadow: none;
                 transition: 300ms ease;
                 transition-property: box-shadow, transform;
             }
 
+            &:focus {
+                background-color: rgba(white, 0.05);
```
Comment (yajo) [inline web_responsive/static/src/css/web_responsive.scss]: The bootstrap theme you use includes a definition of what is "white" and "black", a few shades of gray and all the primary, secondary, complementary colors. It's better to use those, so your addon looks good if there's any theme change.

```suggestion
                background-color: rgba(gray('white'), 0.05);
```

File: web_responsive/static/src/css/web_responsive.scss [hunk 3]
```diff
@@ -210,7 +217,7 @@ $chatter_zone_width: 35%;
 
         .o_app:hover img {
             transform: translateY(-3px);
-            box-shadow: 0 9px 12px -4px rgba(0, 0, 0, 0.3);
+            box-shadow: 0 9px 12px -4px rgba(black, 0.3);
```
Comment (yajo) [inline web_responsive/static/src/css/web_responsive.scss]:
```suggestion
            box-shadow: 0 9px 12px -4px rgba(gray('black'), 0.3);
```

File: web_responsive/static/src/css/web_responsive.scss [hunk 4]
```diff
@@ -227,7 +234,7 @@ $chatter_zone_width: 35%;
             color: white;
             margin-top: 4px;
             font-size: 1.25rem;
-            text-shadow: 1px 1px 1px rgba(0, 0, 0, 0.4);
+            text-shadow: 1px 1px 1px rgba(black, 0.4);
```
Comment (yajo) [inline web_responsive/static/src/css/web_responsive.scss]:
```suggestion
            text-shadow: 1px 1px 1px rgba(gray('black'), 0.4);
```

File: web_responsive/static/src/css/web_responsive.scss [hunk 5]
```diff
@@ -240,20 +247,60 @@ $chatter_zone_width: 35%;
...
+                        color: white;
+                        display: block;
+                        padding: 1px 2px 2px 2px;
+                        box-shadow: none;
+
+                        &::placeholder {
+                            color: white;
+                            opacity: 0.5;
+                        }
+                    }
```
Comment (yajo) [inline web_responsive/static/src/css/web_responsive.scss]: And well, the same here about color variables...

File: web_responsive/static/src/css/web_responsive.scss (hunk 2 above)
Comment (ivantodorovich) [reply web_responsive/static/src/css/web_responsive.scss]: Thanks for the clarification!
Yes, I was aware of the theme-colors, but wasn't aware of the white/black for this case. Even in odoo source code there are lots of cases where the hardcoded keyword black/white is used.  

https://github.com/odoo/odoo/search?l=SCSS&q=white

Comment (yajo) [reply web_responsive/static/src/css/web_responsive.scss]: > Even in odoo source code there are lots of cases where the hardcoded keyword black/white is used

:grin: I know, but well... It's odoo, not OCA :stuck_out_tongue_winking_eye:

**Review by yajo — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please squash commits

**Comment by ivantodorovich** [comment]
> Please squash commits

done

**Review by pedrobaeza — APPROVED** [review]

**Comment by patel1008b** [comment]
New Changes looks great, but something needs to do when its custom theme is used.


Screenshot 2020-08-13 at 6 49 01 PM

we are using a https://apps.odoo.com/apps/themes/13.0/backend_theme_v13/

**Comment by pedrobaeza** [comment]
I think the problem is in the theme itself, not here.

**Comment by androettop** [comment]
> New Changes looks great, but something needs to do when its custom theme is used.
> 
> Screenshot 2020-08-13 at 6 49 01 PM
> 
> we are using a https://apps.odoo.com/apps/themes/13.0/backend_theme_v13/

The theme developer must adapt the theme to the new web_responsive style

**Pull request merged**